spec/models/user_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+require "shared/share_example_spec.rb"
+
+RSpec.describe User, type: :model do
+  subject {FactoryBot.create(:user)} 
+
+  it "has a valid data" do
+    expect(subject).to be_valid
+  end
+
+  describe "associations" do
+    it {is_expected.to have_many(:orders).dependent(:destroy)}
+    it {is_expected.to have_many(:addresses).dependent(:destroy)}
+  end
+
+  describe "#email_downcase" do
```

Comment:
phân biệt describer context và it em nhé

Reply:
Dạ a,
- describer dùng để nhóm các test case theo class, module, method hoặc theo action của controller thành 1 group, bên trong group này có thể là 1 example hoặc nested example. nếu là 1 example thì dùng it, nếu là nested example thì dùng context
- context dùng để nhóm các điều kiện giống nhau của các example.
- it được dùng để định nghĩa 1 example

Reply:
ready